src/editor/components/modals/file-modal/manage-file-sources/add-new-file-source.jsx

```diff
@@ -0,0 +1,225 @@
+import React, { useState, useEffect } from "react";
+import PropTypes from "prop-types";
+import { connect } from "react-redux";
+import styled from "react-emotion";
+
+import TextInput from "./text-input";
+import { ContainedButton } from "../../../../../shared/components/buttons";
+import { addFileSource } from "../../../../actions/file-source-actions";
+import { FILE_SOURCE_UPDATE_SELECTOR_OPTIONS } from "../../../../state-schemas/state-schema";
+
+const AddNewSourceContainer = styled.div`
+  display: grid;
+  grid-template-columns: auto auto auto auto;
+  align-items: start;
+  grid-column-gap: calc(var(--marg) * 2);
+`;
+
+const AddNewSourceButton = styled(ContainedButton)`
+  justify-self: start;
+  margin: 0;
+`;
+
+const FileSourceStatus = styled.div`
+  min-height: 1em;
+  overflow: hidden;
+  text-align: center;
+  margin-top: 10px;
+  margin-bottom: 10px;
+`;
+
+const FileSourceStatusText = styled.div`
+  transform: translateY(${props => (props.isVisible ? "0" : "1.5em")});
+  transition: 500ms;
+  color: ${props => (props.statusType === "ERROR" ? "red" : "black")};
+`;
+
+const UpdateIntervalContainer = styled.div`
+  span {
+    font-size: 13px;
+  }
+`;
+
+const UpdateIntervalSelector = styled.select`
```

Comment:
the styling and behavior of this button don't really match the other buttons in the rest of the app (using a system default `select` menu rather than a custom drop down, the corner radius is different, the text doesn't match the all caps that we have in other buttons... @hamilton, i wonder if we can reuse some of the custom components you worked on earlier this year rather than this one-off code for `UpdateIntervalContainer` (and the stuff supporting it). 

I also notice that the code here with the background image svg seems rather one-off and hard to follow compared to using a MUI icon in a component. i bet we can clean this out with minimal muss and fuss

 (as well as the code here with the background image svg)

Reply:
yep will do